table/block_based/block_based_table_builder.cc

```diff
@@ -1851,7 +1879,7 @@ Status BlockBasedTableBuilder::CompressAndVerifyBlock(
     compressor = r->data_block_compressor;
     verify_decomp = r->data_block_verify_decompressor.get();
   } else {
-    compressor = r->basic_compressor.get();
+    compressor = r->index_block_compressor;
```

Comment:
I'm confused here - when is_data_block=false, is it necessarily index block? If it's not, can you explain more why it's okay to use index_block_compressor? Similarly for "is_data_block ? r->data_block_working_area : r->index_block_working_area,".